src/lib/components/Sidebar.svelte

```diff
@@ -37,6 +47,23 @@
     }
     return typeof interval === 'number' ? interval : titleMap[interval]
   }
+
+  $: getCurrentCategoryKeys = () => {
+    const keys = Object.keys(chatsGroupedByTime)
+    const keysObject: KeyObject = keys.reduce((keysObjectAcc, key) => {
+      keysObjectAcc[key] = ['today', 'yesterday', 'previousSevenDays'].includes(key)
+      return keysObjectAcc
+    }, {} as KeyObject)
+    return keysObject
+  }
+
+  $: currentCategoryState = getCurrentCategoryKeys()
```

Comment:
I think that now it actually starts making sense to move both the open/close state and the label into a common object for a "chat group", like we talked about in the other PR? Everything would be basically the same, but instead of keeping separate arrays for the label and categoryState, you would return an object with them from the categorizeDate function, e.g. `{ key: "today", label: "Today", isOpen: true }`.

Reply:
Then it's easy to add more attributes to each chat group in the future as well 😃